src/lib/input/input.ts

```diff
@@ -368,5 +368,12 @@ export class MatInput extends _MatInputMixinBase implements MatFormFieldControl<
    * Implemented as part of MatFormFieldControl.
    * @docs-private
    */
-  onContainerClick() { this.focus(); }
+  onContainerClick() {
+    // Do not re-focus the input element if the element is already focused. Otherwise it can happen
+    // that someone clicks on a time input and the cursor resets to the "hours" field while the
+    // "minutes" field was actually clicked. See: https://github.com/angular/material2/issues/12849
+    if (!this.focused) {
```

Comment:
Do we need similar logic for other `MatFormFieldControl` components (e.g. select, chips)

Reply:
It guess it wouldn't hurt. There might be also people that use chips with a time input.

I can update it tomorrow (or have a follow-up PR)

Reply:
Just checked. The chip list already has a proper check for that. Please have another look.